qiskit/providers/backend_compat.py

```diff
@@ -0,0 +1,249 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Backend abstract interface for providers."""
+
+
+from typing import List, Union, Iterable, Any, Dict
+
+from qiskit.providers.backend import BackendV1, BackendV2
+from qiskit.transpiler.target import Target, InstructionProperties
+from qiskit.providers.backend import QubitProperties
+from qiskit.utils.units import apply_prefix
+from qiskit.circuit.library.standard_gates import IGate, SXGate, XGate, CXGate, RZGate
+from qiskit.circuit.parameter import Parameter
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.delay import Delay
+from qiskit.circuit.measure import Measure
+from qiskit.circuit.reset import Reset
+from qiskit.providers.models.backendconfiguration import BackendConfiguration
+from qiskit.providers.models.backendproperties import BackendProperties
+from qiskit.providers.models.pulsedefaults import PulseDefaults
+from qiskit.providers.exceptions import BackendPropertyError
+
+
+def convert_to_target(
+    configuration: BackendConfiguration,
+    properties: BackendProperties = None,
+    defaults: PulseDefaults = None,
+) -> Target:
+    """Uses configuration, properties and pulse defaults
+    to construct and return Target class.
+    """
+    name_mapping = {
+        "id": IGate(),
+        "sx": SXGate(),
+        "x": XGate(),
+        "cx": CXGate(),
+        "rz": RZGate(Parameter("λ")),
+        "reset": Reset(),
+    }
+    custom_gates = {}
+    target = None
+    # Parse from properties if it exsits
+    if properties is not None:
+        qubit_properties = qubit_props_list_from_props(properties=properties)
+        target = Target(num_qubits=configuration.n_qubits, qubit_properties=qubit_properties)
+        # Parse instructions
+        gates: Dict[str, Any] = {}
+        for gate in properties.gates:
+            name = gate.gate
+            if name in name_mapping:
+                if name not in gates:
+                    gates[name] = {}
+            elif name not in custom_gates:
+                custom_gate = Gate(name, len(gate.qubits), [])
```

Comment:
This may have parameters, i.e. when a backend defines Rx gate. Is there any mechanism to check parameters? Perhaps you can check the standard library in circuits and extract definition, rather than explicitly separating `custom_gate` from `gates`.

Reply:
Yeah, this isn't perfect, but I'm not quite sure how best to handle the case when we don't know what the gate object is for a name in the configuration. Do you think it would be better to just raise if the `name_mapping` doesn't contain a gate for a name defined in the basis gates?

Reply:
Yeah, let me add an option to add a custom gate mapping and use that too

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8759/commits/7e8fe264bbbc3d7742bf23e973867e7771fb8213